### PR TITLE
Annotate NamespaceOrTypeSymbol

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

**Comment by sharwell** [comment]
@dotnet/roslyn-compiler  for review

File: src/Compilers/CSharp/Portable/Symbols/NamespaceOrTypeSymbol.cs [hunk 1]
```diff
@@ -168,26 +170,26 @@ public virtual ImmutableArray<NamedTypeSymbol> GetTypeMembers(string name, int a
         /// Get a source type symbol for the given declaration syntax.
         /// </summary>
         /// <returns>Null if there is no matching declaration.</returns>
-        internal SourceNamedTypeSymbol GetSourceTypeMember(TypeDeclarationSyntax syntax)
+        internal SourceNamedTypeSymbol? GetSourceTypeMember(TypeDeclarationSyntax syntax)
         {
-            return GetSourceTypeMember(syntax.Identifier.ValueText, syntax.Arity, syntax.Kind(), syntax);
+            return GetSourceTypeMember(syntax.Identifier.ValueText ?? "", syntax.Arity, syntax.Kind(), syntax);
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Symbols/NamespaceOrTypeSymbol.cs]: >syntax.Identifier.ValueText ?? "" [](start = 39, length = 33)

Same comment as in another PR. This doesn't feel right.
Same comment a few lines below

Comment (sharwell) [reply src/Compilers/CSharp/Portable/Symbols/NamespaceOrTypeSymbol.cs]: This is fixed now.

**Comment by jcouv** [comment]
@333fred for a second review (tiny PR, trying to clean up our backlog of reviews). Thanks

**Comment by jcouv** [comment]
@sharwell This can be merged. Thanks

**Pull request merged**